services/brig/src/Brig/Run.hs

```diff
@@ -103,3 +124,51 @@ lookupRequestIdMiddleware :: (RequestId -> Wai.Application) -> Wai.Application
 lookupRequestIdMiddleware mkapp req cont = do
   let reqid = maybe def RequestId $ lookupRequestId req
   mkapp reqid req cont
+
+cleanExpiredPendingInvitations :: AppIO ()
+cleanExpiredPendingInvitations = do
+  safeForever "cleanExpiredPendingInvitations" $ do
+    today <- utctDay <$> (liftIO =<< view currentTime)
+    for_ [0 .. 9] $ \i ->
+      cleanUpDay (addDays (- i) today)
+    let d :: Int = 24 * 60 * 60
+    randomSecs <- liftIO (round <$> randomRIO @Double (0.5 * fromIntegral d, fromIntegral d))
+    threadDelay (randomSecs * 1_000_000)
+  where
+    cleanUpDay :: Day -> AppIO ()
+    cleanUpDay day = do
+      expiredEntries <- forTrackedExpirations day $ \exps ->
+        catMaybes
+          <$> ( for exps $ \(PendingActivationExpiration uid expiresAt tid) -> do
+                  isExpired <- (expiresAt <=) <$> (liftIO =<< view currentTime)
+                  if isExpired
+                    then do
+                      isPendingInvitation <- (Just PendingInvitation ==) <$> lookupStatus uid
+                      invExpired <- isNothing <$> Data.lookupInvitation tid (coerce uid)
+                      when (isPendingInvitation && invExpired) $ do
+                        API.deleteUserNoVerify uid
```

Comment:
Yeah, we should keep an eye on this.

I suggest we:

- add a function `deleteUsersNoVerify` that deletes many users
- we can implement it for now using `mapM_`
- but we should call it only once per page of `PendingActivationExpiration`
- and keep a metric that tells us how often this is called
- ideally, we should also have a metric that tells us how much time & space deletion takes, but that may be too much work for now.

Reply:
To avoid scheduling to many deletions to fast, I added a throttling delay. With at most 2 items per second (that 170k per day).